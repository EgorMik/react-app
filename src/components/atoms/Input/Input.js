import React from 'react';
import PropTypes from 'prop-types';
import styles from './Input.module.scss';

const Input = ({ value, updateText, radio, handleRadio}) => {
    return (
    <div className ={styles.header}>

      <input 
      type="checkbox" 
      onClick={handleRadio}
      onChange={() => {}} checked={radio}
      className={styles.checkbox}/>

      <input 
      type="text" 
      placeholder="Add your important task"
      value={value}
      onChange={(e) => updateText(e.target.value)}
      className={styles.input}/>

    </div>
    );
  };
  Input.propTypes = {
    value: PropTypes.string,
    updateText: PropTypes.func,
    radio: PropTypes.bool,
    handleRadio: PropTypes.func,
  }
  export default Input;