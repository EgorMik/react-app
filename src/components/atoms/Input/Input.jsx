import React from 'react';
import PropTypes from 'prop-types';

const Input = ({ value, updateText, radio, handleRadio}) => {
    return (
      <div className ='radioInput'>
        <li>
      <em onClick={handleRadio} className={ radio ? 'selected' : null}></em>
      </li>
        <input type="text" placeholder="Add your important task"
          value={value}
          onChange={(e) => updateText(e.target.value)}/>
        
        </div>
    );
  };
  Input.propTypes = {
    value: PropTypes.string,
    updateText: PropTypes.func,
    radio: PropTypes.bool,
    handleRadio: PropTypes.func,
  }
  export default Input;