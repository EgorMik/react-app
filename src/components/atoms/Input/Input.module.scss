$h: 10px;
$color: rgb(58, 59, 59);
$width: 200;
$gradient: linear-gradient(to right, rgb(95, 95, 223), rgb(211, 125, 225));
@mixin boxSizing {
    box-sizing: border-box;
    -moz-box-sizing: border-box;
    -webkit-box-sizing: border-box;
}

@mixin transform($transforms) {
    -moz-transform: $transforms;
    -o-transform: $transforms;
    -ms-transform: $transforms;
    -webkit-transform: $transforms;
    transform: $transforms;
}

@mixin rotate($deg) {
    @include transform(rotate(#{$deg}deg));
}

@mixin border($borderWidth:1px,$borderColor:#d9d9d9,$borderStyle:solid) {
    border: $borderWidth $borderColor $borderStyle;
}

.header {
    padding: 10;
    margin: 40;
    border-radius: 0.8rem;
    
}

.checkbox {
    margin: 9px 4px;
    position: absolute;
    z-index: 100;
    float:left;
    appearance: none;
    border: 0.5px solid #ccc;
    border-radius: 50%;
    width: 20px;
    height: 20px;
    margin-right: 10px;
    cursor: pointer;
    margin-top:9px;
}

.checkbox:checked {
    position: absolute;
    border: 1px solid $gradient;
    background: $gradient;
    &:before {
        display: block;
        content: '';
        width: 6px;
        height: 2px;
        background: white;
        position: absolute;
        left: 4.5px;
        top: 10.6px;
        @include rotate(45)
    }
    &:after {
        display: block;
        content: '';
        width: 10px;
        height: 2px;
        background: white;
        position: absolute;
        left: 8px;
        top: 8px;
        @include rotate(-45)
    }
}

input[type=text] {
    font-family: 'Josefin Sans', sans-serif;
    height: $h*4;
    @include border;
    background: #fff;
    width: $width*2+163+px;
    padding: 0 5px;
    border-radius: 0.8rem;
    position: relative;
    padding-left: 31px;
}