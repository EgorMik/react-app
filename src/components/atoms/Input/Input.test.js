import React from "react";
import renderer from "react-test-renderer";
import Input from "./Input";

describe("Input", () => {
  it("should renders correctly", () => {
    const tree = renderer.create(<Input name="test" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});