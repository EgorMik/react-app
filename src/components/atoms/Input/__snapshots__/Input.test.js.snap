// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should renders correctly 1`] = `
<div
  className="header"
>
  <input
    className="checkbox"
    onChange={[Function]}
    type="checkbox"
  />
  <input
    className="input"
    onChange={[Function]}
    placeholder="Add your important task"
    type="text"
  />
</div>
`;
